### DISCUSSION TITLE
Rancher Desktop, WSL and docker

### DISCUSSION THREAD

**Discussion by flexguse** [rancher-sandbox/rancher-desktop#1215] [Ideas]
First of all I would like to thank everybody for Rancher Desktop 0.7.1! It's awesome I can now use docker CLI in Windows!

I would like to replace Docker for Desktop as my development kubernetes environment. Rancher Desktop 0.7.1 with docker in it is a huge step forward but in my case I need docker integration in WSL, too.

So what  I need is:

- install Rancher Desktop
- enable WSL2 Integration
- ability to run docker cli in Windows and WSL
- ability to run kubectl cli in Windows and WSL

Did I miss something, or is this already included in Rancher Desktop?
If not, are there any plans to enhance the WSL integration by docker CLI?

Regards,
Christoph

**Comment by mook-as** [comment]
Hi! This is meant to be included already (using the moby backend may provide better compatibility, as in that case we just use the same CLI code).  If you enable WSL integration in Rancher Desktop, you should see a process in your WSL distribution (`wsl-helper docker-proxy serve` from the Rancher Desktop installation directory), which provides `/var/run/docker.sock`.  We also update `~/.kube/config` for kubectl compatibility.

On the Windows side, things are similar — `wsl-helper.exe docker-proxy serve` provides the `\\.\pipe\docker_engine` named pipe, and `%USERPROFILE%\.kube\config` is updated.

**Comment by flexguse** [comment]
Hi! Thanks a lot for your answer.

In the Ubuntu WSL I found /var/run/docker.sock so I installed the docker-ce-cli et voilà the docker-cli worked like a charm.
